client/src/views/GroupContainer/GroupListView.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { Route, BrowserRouter, Switch } from 'react-router-dom';
+import WelcomeView from './WelcomeView.jsx';
+import GroupList from './GroupList.jsx';
+import GroupItem from './GroupItem.jsx';
+import PostItActionTypes from '../../data/PostItActionTypes';
+import PostItDispatcher from '../../data/PostItDispatcher';
+import bulkMessageRequest from '../../utility/bulkMessageRequest';
+
+/**
+ * GroupList is a container for the list of groups, also doubles as a navlink
+ * @param {*} props
+ * @return {void}
+ */
+function GroupListView(props) {
+  const socketProp = props.socket;
+
+  // attach listener for new messages
+  socketProp.on('newMessage', (newMessages) => {
+    PostItDispatcher.handleServerAction({
+      type: PostItActionTypes.RECIEVE_MESSAGE_RESPONSE,
+      Id: newMessages.Id,
```

Comment:
action should comprise of `type` and the payload object which you can name anything , the payload should be an object with all these other variables inside